crates/holochain/src/core/workflow/countersigning_workflow/incomplete.rs

```diff
@@ -0,0 +1,336 @@
+                                            SessionCompletionDecision::Abandoned
+                                        }
+                                    }
+                                    RecordEntry::Present(_) => {
+                                        // Anything else on the chain is evidence that the agent did not complete the countersigning.
+                                        tracing::debug!(
+                                            "Agent {:?} has a non-countersigning entry",
+                                            agent
+                                        );
+                                        SessionCompletionDecision::Abandoned
+                                    }
+                                    RecordEntry::Hidden | RecordEntry::NA => {
+                                        // This wouldn't be the case for a countersigning entry so this is evidence that
+                                        // the agent has put something else on their chain.
+                                        SessionCompletionDecision::Abandoned
+                                    }
+                                    RecordEntry::NotStored => {
+                                        // This case is not determinate. The agent activity authority might
+                                        // have the action but not the entry yet. We can't make a decision here.
+                                        // In this case, we will also have tried to ask a record authority for
+                                        // this missing entry.
+                                        SessionCompletionDecision::Indeterminate
+                                    }
+                                }
+                            }
+                        }
+                        _ => {
+                            tracing::warn!("Agent authority returned an unexpected response for agent {:?}: {:?}", agent, activity);
+                            continue;
+                        }
+                    }
+                }
+                e => {
+                    tracing::debug!(
+                        "Failed to get activity for agent {:?} because of: {:?}",
+                        agent,
+                        e
+                    );
+                    SessionCompletionDecision::Indeterminate
+                }
+            };
+
+            authority_decisions.push(decision);
+        }
+
+        resolution_outcomes.push(SessionResolutionOutcome {
+            agent: agent.clone(),
+            decisions: authority_decisions.clone(),
+        });
+
+        if authority_decisions.len() < NUM_AUTHORITIES_TO_QUERY {
+            // We are requiring all the authorities to agree, so if we don't have enough responses
+            // then we can't make a decision.
+            // That is likely to make the resolution process slower, but it's more likely to be correct.
```

Comment:
```suggestion
            // We are requiring all the authorities to agree, so if we don't have enough responses
            // then we can't make a decision.
            // That is likely to make the resolution process slower, but it's more likely to be correct.
            // NOTE: at the moment, since we're calling `get_agent_activity` without a target,
            //       all the responses could have come from the same authority.
```

Reply:
Yes, very important to note! Thank you